ForgeUnity/Assets/BeardedManStudios/Scripts/NetworkManager.cs

```diff
@@ -72,265 +72,146 @@ protected virtual void OnDisable()
 		}
 
-		public virtual void MatchmakingServersFromMasterServer(string masterServerHost,
-			ushort masterServerPort,
-			int elo,
-			System.Action<MasterServerResponse> callback = null,
-			string gameId = "myGame",
-			string gameType = "any",
+		public virtual void MatchmakingServersFromMasterServer(string masterServerHost, ushort masterServerPort, int elo,
+			System.Action<MasterServerResponse> callback = null, string gameId = "myGame", string gameType = "any",
 			string gameMode = "all")
 		{
-			// The Master Server communicates over TCP
-			TCPMasterClient client = new TCPMasterClient();
-
-			// Once this client has been accepted by the master server it should send it's get request
-			client.serverAccepted += (sender) =>
-			{
-				try
-				{
-					// Create the get request with the desired filters
-					JSONNode sendData = JSONNode.Parse("{}");
-					JSONClass getData = new JSONClass();
-					getData.Add("id", gameId);
-					getData.Add("type", gameType);
-					getData.Add("mode", gameMode);
-					getData.Add("elo", new JSONData(elo));
-
-					sendData.Add("get", getData);
-
-					// Send the request to the server
-					client.Send(Text.CreateFromString(client.Time.Timestep, sendData.ToString(), true, Receivers.Server, MessageGroupIds.MASTER_SERVER_GET, true));
-				}
-				catch
-				{
-					// If anything fails, then this client needs to be disconnected
-					client.Disconnect(true);
-					client = null;
-
-					MainThreadManager.Run(() =>
-					{
-						if (callback != null)
-							callback(null);
-					});
-				}
-			};
-
-			// An event that is raised when the server responds with hosts
-			client.textMessageReceived += (player, frame, sender) =>
+			var fetcher = MasterServerRequestManger.CreateWithData(new MasterServerRequestManger.RequestData
 			{
-				try
-				{
-					// Get the list of hosts to iterate through from the frame payload
-					JSONNode data = JSONNode.Parse(frame.ToString());
-					MainThreadManager.Run(() =>
-					{
-						if (data["hosts"] != null)
-						{
-							MasterServerResponse response = new MasterServerResponse(data["hosts"].AsArray);
-							if (callback != null)
-								callback(response);
-						}
-						else
-						{
-							if (callback != null)
-								callback(null);
-						}
-					});
-				}
-				finally
-				{
-					if (client != null)
-					{
-						// If we succeed or fail the client needs to disconnect from the Master Server
-						client.Disconnect(true);
-						client = null;
-					}
-				}
-			};
-
-			try
-			{
-				client.Connect(masterServerHost, masterServerPort);
-			}
-			catch (System.Exception ex)
-			{
-				Debug.LogError(ex.Message);
-				MainThreadManager.Run(() =>
-				{
-					if (callback != null)
-						callback(null);
-				});
-			}
+				completedCallback = callback,
+				elo = elo,
+				gameId = gameId,
+				gameMode = gameMode,
+				gameType = gameType,
+				hostAddress = masterServerHost,
+				hostPort = masterServerPort
+			});
+			fetcher.SendRegisterRequest();
```

Comment:
I think this method is a bit misleading as we are getting a list of servers from the master server and not registering with it.